### ISSUE TITLE
[iOS] Crash in specific map area

### ISSUE THREAD

**Issue by dvdmrtnz** [organicmaps/organicmaps#6158]
I have built the iPhone app from latest master and the app is crashing when zooming in to an area around here:
https://www.openstreetmap.org/#map=17/42.80912/-1.63816
https://omaps.app/on_T5ETlOL

It happens with builds after commit 5e0a1266199f1107ad0d4feeb0bfbcad7e580330. I have built with older commits and the app works fine. 

The XCode debugger points to this part of the code in `geometry/rect2d.hpp` file when the crash happens:

```
  Rect(T minX, T minY, T maxX, T maxY) : m_minX(minX), m_minY(minY), m_maxX(maxX), m_maxY(maxY)
  {
    ASSERT(minX <= maxX, ());
    ASSERT(minY <= maxY, ());
  }
```
 
Attached is the related crash log taken from Settings > Privacy > Analytics & Improvements > Analytics Data, if it helps.

[Organic Maps (Debug)-2023-09-29-172728.txt](https://github.com/organicmaps/organicmaps/files/12768440/Organic.Maps.Debug.-2023-09-29-172728.txt)

**Comment by dvdmrtnz** [comment]
@pastk This crash happens after changes in PR https://github.com/organicmaps/organicmaps/pull/6034

**Comment by pastk** [comment]
I'll look into this, thanks for the report!

**Comment by pastk** [comment]
Hmm, I can't reproduce it so far...

Are there OM logs you can attach?

**Comment by pastk** [comment]
Maybe your Spain_Comunidad Foral de Navarra map version is different? I've downloaded the latest 230920.

**Comment by dvdmrtnz** [comment]
> Hmm, I can't reproduce it so far...
> 
> Are there OM logs you can attach?

How do I get the OM logs?

**Comment by pastk** [comment]
I've never worked with XCode but there should be a console with app's output somewhere...

**Comment by dvdmrtnz** [comment]
> Maybe your Spain_Comunidad Foral de Navarra map version is different? I've downloaded the latest 230920.

I just deleted the app and let it download the latest map from the server. The crash still happens.



> > Hmm, I can't reproduce it so far...
> > Are there OM logs you can attach?
> 
> How do I get the OM logs?

Oh you mean the console output. Sure, here it is:

```
LOG TID(1) DEBUG     99.1547 drape_frontend/drape_engine.cpp:389 SetRenderingDisabled(): Rendering disabled
LOG TID(1) DEBUG     99.9959 drape_frontend/drape_engine.cpp:381 SetRenderingEnabled(): Rendering enabled
LOG TID(1) DEBUG      173.15 storage/storage.cpp:743 operator()(): Successful SHA check
LOG TID(1) INFO       173.15 storage/storage.cpp:760 operator()(): Registering downloaded file: Spain_Comunidad Foral de Navarra Map ; success: 1
LOG TID(1) INFO      173.151 map/framework.cpp:146 RegisterMap(): Loaded Spain_Comunidad Foral de Navarra map, of version 230920
LOG TID(7) DEBUG     173.459 drape_frontend/metaline_manager.cpp:69 OnTaskFinished(): Metalines prepared: MwmId [Spain_Comunidad Foral de Navarra, 230920]
TID(4) ASSERT FAILED
geometry/rect2d.hpp:45
CHECK(minX <= maxX) 
Assertion failed: (false), function Rect, file rect2d.hpp, line 45.
```
The variables watch shows:

```
this	m2::Rect<double> *	0x16db72010	0x000000016db72010
m_minX	double	NaN
m_minY	double	NaN
m_maxX	double	NaN
m_maxY	double	NaN
minX	double	NaN
minY	double	NaN
maxX	double	NaN
maxY	double	NaN

```

**Comment by pastk** [comment]
This assert has fired once for me too at this spot so far (out of many re-starts)! 
So for me its random and rare, maybe some uninitialized variable..
Does the crash happen for you every time?

**Comment by pastk** [comment]
Could you please replace those two asserts in rect2d.hpp with a single
`ASSERT(minX <= maxX && minY <= maxY, (minX, maxX, minY, maxY));`
and re-run?
I hope there will be feature coords in the console output after that :)

**Comment by dvdmrtnz** [comment]
> This assert has fired once for me too at this spot so far (out of many re-starts)! So for me its random and rare, maybe some uninitialized variable.. Does the crash happen for you every time?

It happens every single time when panning to this area.



> Could you please replace those two asserts in rect2d.hpp with a single `ASSERT(minX <= maxX && minY <= maxY, (minX, maxX, minY, maxY));` and re-run? I hope there will be feature coords in the console output after that :)

With that change: 

```
LOG TID(3) WARNING   3.71937 drape/texture_manager.cpp:562 GetSymbolRegion(): Detected using of unknown symbol  transit_air_service-s
LOG TID(3) WARNING   3.71937 drape/texture_manager.cpp:562 GetSymbolRegion(): Detected using of unknown symbol  transit_air_service-m
LOG TID(3) WARNING   3.71938 drape/texture_manager.cpp:562 GetSymbolRegion(): Detected using of unknown symbol  transit_water_service-s
LOG TID(3) WARNING   3.71939 drape/texture_manager.cpp:562 GetSymbolRegion(): Detected using of unknown symbol  transit_water_service-m
LOG TID(2) DEBUG     4.14063 search/cities_boundaries_table.cpp:104 Load(): Localities count = 119114 ; with boundary = 45909
LOG TID(2) INFO      4.14734 search/processor.cpp:340 LoadCitiesBoundaries(): Loaded cities boundaries
LOG TID(5) DEBUG     4.35585 drape_frontend/metaline_manager.cpp:69 OnTaskFinished(): Metalines prepared: MwmId [Spain_Comunidad Foral de Navarra, 230920]
TID(4) ASSERT FAILED
geometry/rect2d.hpp:45
CHECK(minX <= maxX && minY <= maxY) nan nan nan nan
Assertion failed: (false), function Rect, file rect2d.hpp, line 45.
```

**Comment by dvdmrtnz** [comment]
It crashes in simulator too:


https://github.com/organicmaps/organicmaps/assets/47610359/b6690106-307b-4fa6-8543-bf3d3df5f909

**Comment by biodranik** [comment]
@dvdmrtnz what's in the call stack? Who produces invalid rect with NaNs ?

**Comment by dvdmrtnz** [comment]
> @dvdmrtnz what's in the call stack? Who produces invalid rect with NaNs ?

```
#0	0x000000010a250a4c in __pthread_kill ()
#1	0x000000010a93b1d0 in pthread_kill ()
#2	0x00000001801375ec in abort ()
#3	0x0000000180136a6c in __assert_rtn ()
#4	0x00000001031ccbe8 in m2::Rect<double>::Rect(double, double, double, double) at /Users/davidmartinez/git/organicmaps/geometry/rect2d.hpp:45
#5	0x00000001031cbe18 in m2::Rect<double>::Rect(double, double, double, double) at /Users/davidmartinez/git/organicmaps/geometry/rect2d.hpp:44
#6	0x00000001032a6efc in df::(anonymous namespace)::StraightTextHandle::GetPixelRect(ScreenBase const&, bool) const at /Users/davidmartinez/git/organicmaps/drape_frontend/text_shape.cpp:129
#7	0x0000000102fed090 in dp::OverlayHandle::GetExtendedPixelRect(ScreenBase const&) const at /Users/davidmartinez/git/organicmaps/drape/overlay_handle.cpp:131
#8	0x0000000103132834 in dp::OverlayTree::Add(ref_ptr<dp::OverlayHandle>) at /Users/davidmartinez/git/organicmaps/drape/overlay_tree.cpp:200
#9	0x0000000102ffee10 in dp::RenderBucket::CollectOverlayHandles(ref_ptr<dp::OverlayTree>) at /Users/davidmartinez/git/organicmaps/drape/render_bucket.cpp:74
#10	0x00000001033d3174 in df::RenderGroup::CollectOverlay(ref_ptr<dp::OverlayTree>) at /Users/davidmartinez/git/organicmaps/drape_frontend/render_group.cpp:47
#11	0x000000010348c578 in df::FrontendRenderer::UpdateOverlayTree(ScreenBase const&, std::__1::unique_ptr<df::RenderGroup, std::__1::default_delete<df::RenderGroup>>&) at /Users/davidmartinez/git/organicmaps/drape_frontend/frontend_renderer.cpp:1379
#12	0x000000010348be50 in df::FrontendRenderer::BuildOverlayTree(ScreenBase const&) at /Users/davidmartinez/git/organicmaps/drape_frontend/frontend_renderer.cpp:1866
#13	0x000000010348ebb8 in df::FrontendRenderer::RenderOverlayLayer(ScreenBase const&) at /Users/davidmartinez/git/organicmaps/drape_frontend/frontend_renderer.cpp:1597
#14	0x000000010348ced4 in df::FrontendRenderer::RenderScene(ScreenBase const&, bool) at /Users/davidmartinez/git/organicmaps/drape_frontend/frontend_renderer.cpp:1472
#15	0x0000000103490b78 in df::FrontendRenderer::RenderFrame() at /Users/davidmartinez/git/organicmaps/drape_frontend/frontend_renderer.cpp:1766
#16	0x00000001031926bc in df::BaseRenderer::IterateRenderLoopImpl() at /Users/davidmartinez/git/organicmaps/drape_frontend/base_renderer.cpp:64
#17	0x0000000103194c80 in df::BaseRenderer::IterateRenderLoop()::$_0::operator()() const at /Users/davidmartinez/git/organicmaps/drape_frontend/base_renderer.cpp:59
#18	0x0000000103194c58 in decltype(std::declval<df::BaseRenderer::IterateRenderLoop()::$_0&>()()) std::__1::__invoke[abi:v160006]<df::BaseRenderer::IterateRenderLoop()::$_0&>(df::BaseRenderer::IterateRenderLoop()::$_0&) at /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/__functional/invoke.h:394
#19	0x0000000103194c10 in void std::__1::__invoke_void_return_wrapper<void, true>::__call<df::BaseRenderer::IterateRenderLoop()::$_0&>(df::BaseRenderer::IterateRenderLoop()::$_0&) at /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/__functional/invoke.h:487
#20	0x0000000103194bec in std::__1::__function::__alloc_func<df::BaseRenderer::IterateRenderLoop()::$_0, std::__1::allocator<df::BaseRenderer::IterateRenderLoop()::$_0>, void ()>::operator()[abi:v160006]() at /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/__functional/function.h:185
#21	0x0000000103193a60 in std::__1::__function::__func<df::BaseRenderer::IterateRenderLoop()::$_0, std::__1::allocator<df::BaseRenderer::IterateRenderLoop()::$_0>, void ()>::operator()() at /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/__functional/function.h:356
#22	0x0000000102f1b6c8 in std::__1::__function::__value_func<void ()>::operator()[abi:v160006]() const at /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/__functional/function.h:510
#23	0x0000000102f19434 in std::__1::function<void ()>::operator()() const at /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/__functional/function.h:1156
#24	0x0000000103079e74 in dp::RenderFrameMediator(std::__1::function<void ()>&&) at /Users/davidmartinez/git/organicmaps/drape/metal/metal_base_context.mm:424
#25	0x0000000103192608 in df::BaseRenderer::IterateRenderLoop() at /Users/davidmartinez/git/organicmaps/drape_frontend/base_renderer.cpp:59
#26	0x00000001034946d4 in df::FrontendRenderer::Routine::Do() at /Users/davidmartinez/git/organicmaps/drape_frontend/frontend_renderer.cpp:2432
#27	0x0000000102f18f9c in threads::(anonymous namespace)::RunRoutine(std::__1::shared_ptr<threads::IRoutine>) at /Users/davidmartinez/git/organicmaps/base/thread.cpp:26
#28	0x0000000102f1aae8 in decltype(std::declval<void (*)(std::__1::shared_ptr<threads::IRoutine>)>()(std::declval<std::__1::shared_ptr<threads::IRoutine>>())) std::__1::__invoke[abi:v160006]<void (*)(std::__1::shared_ptr<threads::IRoutine>), std::__1::shared_ptr<threads::IRoutine>>(void (*&&)(std::__1::shared_ptr<threads::IRoutine>), std::__1::shared_ptr<threads::IRoutine>&&) at /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/__functional/invoke.h:394
#29	0x0000000102f1aa60 in void std::__1::__thread_execute[abi:v160006]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (*)(std::__1::shared_ptr<threads::IRoutine>), std::__1::shared_ptr<threads::IRoutine>, 2ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (*)(std::__1::shared_ptr<threads::IRoutine>), std::__1::shared_ptr<threads::IRoutine>>&, std::__1::__tuple_indices<2ul>) at /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/thread:288
#30	0x0000000102f1a2b0 in void* std::__1::__thread_proxy[abi:v160006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (*)(std::__1::shared_ptr<threads::IRoutine>), std::__1::shared_ptr<threads::IRoutine>>>(void*) at /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/usr/include/c++/v1/thread:299
#31	0x000000010a93b4c0 in _pthread_start ()
```

**Comment by biodranik** [comment]
@pastk A classical mistake: not initialized ApplyPointFeature::m_centerPoint variable. @dvdmrtnz thanks for the help!

**Comment by vng** [comment]
Fix: https://github.com/organicmaps/organicmaps/pull/6151/commits/b1fe81591144d2fb5d575f7f8d1b66dff6d3c8ba

**Comment by dvdmrtnz** [comment]
> Fix: [b1fe815](https://github.com/organicmaps/organicmaps/commit/b1fe81591144d2fb5d575f7f8d1b66dff6d3c8ba)

Thanks! That fixes it!

**Comment by pastk** [comment]
> Fix: [b1fe815](https://github.com/organicmaps/organicmaps/commit/b1fe81591144d2fb5d575f7f8d1b66dff6d3c8ba)

It looks good, thanks!

**Comment by pastk** [comment]
Fixed in #6151
